htlcswitch/link.go

```diff
@@ -1057,7 +1044,21 @@ func (l *channelLink) htlcManager() {
 	// the channel is not pending, otherwise we should have no htlcs to
 	// reforward.
 	if l.ShortChanID() != hop.Source {
-		if err := l.resolveFwdPkgs(); err != nil {
+		err := l.resolveFwdPkgs()
+		switch err {
+		// No error was encountered, success.
+		case nil:
+
+		// If the duplicate keystone error was encountered, we'll fail
+		// without sending an Error message to the peer.
+		case ErrDuplicateKeystone:
```

Comment:
AFAICT, this'll still cause some implementations like CL to trigger a channel force close which is what we want to avoid in the first place (leads to force closes). 

Seems the root cause of this are duplicate internal retransmissions?

Reply:
This means we won't send Error

Reply:
The root cause is switch-initiated payments that aren't deleted from the circuit map, so this is fixed by passing them to the mailbox. I suppose this bit doesn't actually need to be changed since it shouldn't be hit except for existing buggy channels and you might as well force close as there's no guarantee that lnd can recover.

Reply:
> This means we won't send Error

Ah that's right, we'll special case the errors to decide which one we want to send out: 
https://github.com/lightningnetwork/lnd/blob/ac5f88bae8ca045ce78686c0694ed623d0736a55/htlcswitch/linkfailure.go#L102-L125